### PR TITLE
VPC endpoint to unirpc

### PR DESCRIPTION
**Pull request by melodylove-uniswap**
1. put routing-api behind a vpc
2. created a vpc endpoint to connect to unirpc vpc endpoint service in uniswap-backend account
3. created route53 to point routing-dev to the vpc endpoint to avoid updating the endpoint on the service side.

### CODE REVIEW

File: bin/stacks/routing-lambda-stack.ts [hunk 1]
```diff
@@ -171,6 +244,8 @@ export class RoutingLambdaStack extends cdk.NestedStack {
       runtime: aws_lambda.Runtime.NODEJS_18_X,
       entry: path.join(__dirname, '../../lib/handlers/index.ts'),
       handler: 'quoteHandler',
+      vpc: vpc,
```
Comment (jsy1218) [inline bin/stacks/routing-lambda-stack.ts:247]: is it possible to set vpc to only caching routing lambda first?

Comment (melodylove-uniswap) [reply bin/stacks/routing-lambda-stack.ts:248]: updated to put both lambda under vpc as we discussed

**Review by jsy1218 — APPROVED** [review]

File: bin/stacks/routing-lambda-stack.ts [hunk 2]
```diff
@@ -77,16 +87,79 @@ export class RoutingLambdaStack extends cdk.NestedStack {
...
+          subnetType: ec2.SubnetType.PUBLIC, // Public subnet with internet access
+          cidrMask: 24, // IP range for public subnet
+        },
+        {
+          name: `RoutingPrivateSubnet-${stage}`,
+          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS, // Private subnet with access to the internet via NAT Gateway
+          cidrMask: 24, // IP range for private subnet
+        },
+      ],
+      natGateways: 1, // One NAT Gateway for private subnet internet access
```
Comment (xrsv) [inline bin/stacks/routing-lambda-stack.ts:107]: not sure if related but good to mention here:
on the unirpc side we started hitting a port connections limit once i started ramping up traffic for routing api
It's weird that routing api lambda didn't hit similar issue, as it is doing similar size of http calls to QN directly.
After further digging, I see that routing api default VPC doesn't have a NAT gateway (and can't see active connections dashboards)?
But with this change we are introducing new VPC with NAT gateway, so trying to understand if that might cause issues due to high amount of traffic.
Anyway we could test the new vpc introduction somehow before fully enabling?

**Review by xrsv — APPROVED** [review]
thanks - @jsy1218 can help monitor pipeline
as you mentioned, let's block prod deployment in order to test in staging properly

**Comment by melodylove-uniswap** [comment]
added subnets to the vpc to get around the http port issue.

**Pull request merged**